preshed/bloom.pyx

```diff
@@ -54,40 +58,125 @@ cdef class BloomFilter:
 
 
 cdef bytes bloom_to_bytes(const BloomStruct* bloom):
-    py = array("L")
-    py.append(bloom.hcount)
-    py.append(bloom.length)
-    py.append(bloom.seed)
-    for i in range(bloom.length // sizeof(key_t)):
-        py.append(bloom.bitfield[i])
-    if hasattr(py, "tobytes"):
-        return py.tobytes()
-    else:
-        # Python 2 :(
-        return py.tostring()
+    cdef key_t pad = 0 # to differentiate new from old data format
+    cdef key_t version = 1 # hardcoded, can be incremented
+    prefix = struct.pack("<QQQQQ", pad, version, bloom.hcount, bloom.length, bloom.seed)
+    # note that the modulus check is only required for data that has come from
+    # legacy deserialization - otherwise length is always a multiple of
+    # KEY_BITS.
+    buflen = bloom.length // KEY_BITS
+    if bloom.length % KEY_BITS > 0:
+        buflen += 1
```

Comment:
I use this pattern for ceiling division in several places, but I guess upside-down floor division would be better? That would look like this:

```
buflen = -(bloom.length // -KEY_BITS)
```

It's more terse, and perhaps faster, but also rather weird looking.